tests/test_pickle.py

```diff
@@ -19,6 +19,9 @@ def test_pickle_round_trip(cls, coords):
     data = dumps(geom1, HIGHEST_PROTOCOL)
     geom2 = loads(data)
     assert geom2.has_z == geom1.has_z
+    # TODO(shapely-2.0) LinearRing type not preserved in roundtrip
```

Comment:
We were just discussing this in PyGEOS as well -> https://github.com/pygeos/pygeos/pull/190 

There we thought that it might be OK enough to get a LineString back from pickling a LinearRing. However, having the pickling on the python side (still using WKB), we could more easily track the original class as well and ensure it is correctly preserved. 
However, having it in python compared to in the base C extension type has a performance penalty.

Reply:
This is tracked by https://github.com/Toblerity/Shapely/issues/988